src/components/users.js

```diff
@@ -29,12 +31,108 @@ import {
   regex,
   useTranslate,
   Pagination,
+  Button,
+  useNotify,
+  useUnselectAll,
+  required,
+  fetchStart,
+  fetchEnd,
 } from "react-admin";
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogContentText from "@material-ui/core/DialogContentText";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import dataProvider from "../synapse/dataProvider.js";
 
 const UserPagination = props => (
   <Pagination {...props} rowsPerPageOptions={[10, 25, 50, 100, 500, 1000]} />
 );
 
+const ServernoticesButton = ({ record, selectedIds, method = "single" }) => {
+  const [open, setOpen] = useState(false);
+  const notify = useNotify();
+  const unselectAll = useUnselectAll();
+  const translate = useTranslate();
+  const handleDialogOpen = () => setOpen(true);
+  const handleDialogClose = () => setOpen(false);
+  const handleConfirm = values => {
+    fetchStart();
+
+    if (method === "multi") {
```

Comment:
Instead of a separate prop, I would check if selectedIds is an array or a single user.